### PR TITLE
refactor(shorebird_cli): move test mocks and fakes into their own files

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

This is to reduce the tedium of adding new tests.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/test/src/doctor_test.dart [hunk 1]
```diff
@@ -6,11 +6,11 @@ import 'package:shorebird_cli/src/logger.dart';
 import 'package:shorebird_cli/src/validators/validators.dart';
 import 'package:test/test.dart';
 
-class _MockLogger extends Mock implements Logger {}
+class MockLogger extends Mock implements Logger {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/doctor_test.dart]: Shouldn't these be imported from mocks as well?

File: packages/shorebird_cli/test/src/git_test.dart [hunk 2]
```diff
@@ -7,9 +7,9 @@ import 'package:shorebird_cli/src/git.dart';
 import 'package:shorebird_cli/src/process.dart';
 import 'package:test/test.dart';
 
-class _MockShorebirdProcess extends Mock implements ShorebirdProcess {}
+class MockShorebirdProcess extends Mock implements ShorebirdProcess {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/git_test.dart]: same here

File: packages/shorebird_cli/test/src/gradlew_test.dart [hunk 3]
```diff
@@ -11,13 +11,13 @@ import 'package:shorebird_cli/src/platform.dart';
 import 'package:shorebird_cli/src/process.dart';
 import 'package:test/test.dart';
 
-class _MockJava extends Mock implements Java {}
+class MockJava extends Mock implements Java {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/gradlew_test.dart]: same here

File: packages/shorebird_cli/test/src/http_client/retrying_client_test.dart [hunk 4]
```diff
@@ -6,12 +6,12 @@ import 'package:mocktail/mocktail.dart';
 import 'package:shorebird_cli/src/http_client/retrying_client.dart';
 import 'package:test/test.dart';
 
-class _MockHttpClient extends Mock implements http.Client {}
+class MockHttpClient extends Mock implements http.Client {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/http_client/retrying_client_test.dart]: same here

File: packages/shorebird_cli/test/src/ios_deploy_test.dart [hunk 5]
```diff
@@ -14,21 +14,21 @@ import 'package:shorebird_cli/src/process.dart';
 import 'package:shorebird_cli/src/shorebird_env.dart';
 import 'package:test/test.dart';
 
-class _MockLogger extends Mock implements Logger {}
+class MockLogger extends Mock implements Logger {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/ios_deploy_test.dart]: same here

File: packages/shorebird_cli/test/src/java_test.dart [hunk 6]
```diff
@@ -9,9 +9,9 @@ import 'package:shorebird_cli/src/java.dart';
 import 'package:shorebird_cli/src/platform.dart';
 import 'package:test/test.dart';
 
-class _MockAndroidStudio extends Mock implements AndroidStudio {}
+class MockAndroidStudio extends Mock implements AndroidStudio {}
```
Comment (felangel) [inline packages/shorebird_cli/test/src/java_test.dart]: same here

File: packages/shorebird_cli/test/src/git_test.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/git_test.dart]: Fixed

File: packages/shorebird_cli/test/src/gradlew_test.dart (hunk 3 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/gradlew_test.dart]: Fixed

File: packages/shorebird_cli/test/src/http_client/retrying_client_test.dart (hunk 4 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/http_client/retrying_client_test.dart]: Fixed.

File: packages/shorebird_cli/test/src/ios_deploy_test.dart (hunk 5 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/ios_deploy_test.dart]: Fixed

File: packages/shorebird_cli/test/src/java_test.dart (hunk 6 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/java_test.dart]: Fixed

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/test/src/doctor_test.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/doctor_test.dart]: Done

**Comment by eseidel** [comment]
❌ test/src/commands/preview_command_test.dart: PreviewCommand ios exits with code 0 when install/launch succeeds (failed)

**Comment by bryanoltman** [comment]
@eseidel yeah, I think that's a flaky test. Filed https://github.com/shorebirdtech/shorebird/issues/1306.

**Pull request merged**